Gems/Atom/Tools/ShaderManagementConsole/Code/Source/Window/ShaderManagementConsoleWindow.cpp

```diff
@@ -37,6 +38,62 @@ namespace ShaderManagementConsole
         Base::OnDocumentOpened(documentId);
         m_documentInspector->SetDocumentId(documentId);
     }
+
+    void ShaderManagementConsoleWindow::UpdateMenus(QMenuBar* menuBar)
+    {
+        Base::UpdateMenus(menuBar);
+        m_actionSaveAsChild->setVisible(false);
+    }
+
+    AZStd::string ShaderManagementConsoleWindow::GetSaveDocumentParams(const AZStd::string& initialPath) const
+    {
+        // Get shader file path
+        AZStd::string shaderFullPath;
+        AZ::RPI::ShaderVariantListSourceData shaderVariantList = {};
+        ShaderManagementConsoleDocumentRequestBus::EventResult(
+            shaderVariantList,
+            GetCurrentDocumentId(),
+            &ShaderManagementConsoleDocumentRequestBus::Events::GetShaderVariantListSourceData);
+        shaderFullPath = AZ::RPI::AssetUtils::ResolvePathReference(initialPath, shaderVariantList.m_shaderFilePath);
+        
+        QMessageBox msgBox;
+        msgBox.setText("Where do you want to save the list?");
+        QPushButton* projectBtn = msgBox.addButton(QObject::tr("Save to project"), QMessageBox::YesRole);
+        QPushButton* engineBtn = msgBox.addButton(QObject::tr("Save to engine"), QMessageBox::NoRole);
+        msgBox.exec();
+
+        AZStd::string result;
+        if (msgBox.clickedButton() == projectBtn)
+        {
+            AZ::IO::FixedMaxPath projectPath = AZ::Utils::GetProjectPath();
+            AZStd::string relativePath, rootFolder;
+            bool pathFound;
+            AzToolsFramework::AssetSystemRequestBus::BroadcastResult(
+                pathFound,
+                &AzToolsFramework::AssetSystemRequestBus::Events::GenerateRelativeSourcePath,
+                shaderFullPath,
+                relativePath,
+                rootFolder);
```

Comment:
The `bool pathFound` variable must be initialized with false due to EBus calls not necessarily having any handlers attached to it.

If there are no handlers attached, then the pathFound variable will be left in an indeterminate state because it was not initialized.

```suggestion
            AZ::IO::Path relativePath, rootFolder;
            bool pathFound = false;
            AzToolsFramework::AssetSystemRequestBus::BroadcastResult(
                pathFound,
                &AzToolsFramework::AssetSystemRequestBus::Events::GenerateRelativeSourcePath,
                shaderFullPath.Native(),
                relativePath.Native(),
                rootFolder.Native());
```

Reply:
Good catch, thanks!